### PR TITLE
[IMP] stock_picking_partner_note check note already in use

### PR DESCRIPTION
**Pull request by santostelmo**
Add setting option to validate if picking note is already used in other partners, preventing it to be modified or deleted,

### CODE REVIEW

File: stock_picking_partner_note/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    check_note_already_in_use = fields.Boolean(
+        string="Check picking note already in use",
```
Comment (rousseldenis) [inline stock_picking_partner_note/models/res_company.py:11]: @santostelmo You should explain in which flows (update / delete) - Update README too

File: stock_picking_partner_note/models/res_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    check_note_already_in_use = fields.Boolean(
+        related="company_id.check_note_already_in_use",
+        readonly=False,
+        store=True,
```
Comment (rousseldenis) [inline stock_picking_partner_note/models/res_config_settings.py]: store not necessary

File: stock_picking_partner_note/models/stock_picking_note.py [hunk 3]
```diff
@@ -14,3 +15,28 @@ class StockPickingNote(models.Model):
     active = fields.Boolean(default=True)
     note_type_id = fields.Many2one("stock.picking.note.type", required=True)
     sequence = sequence = fields.Integer(related="note_type_id.sequence", store=True)
+
+    def write(self, vals):
+        if self.env.user.company_id.check_note_already_in_use:
+            self.check_note_already_in_use()
+        return super().write(vals)
+
+    def unlink(self):
```
Comment (rousseldenis) [inline stock_picking_partner_note/models/stock_picking_note.py]: FYI, you can also use @api.ondelete.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

@santostelmo could you also add some tests ?

File: stock_picking_partner_note/models/res_company.py (hunk 1 above)
Comment (santostelmo) [reply stock_picking_partner_note/models/res_company.py:11]: Added some help and a line in readme

**Comment by santostelmo** [comment]
@rousseldenis  Thanks for the review.
I updated the code with your change requests

File: stock_picking_partner_note/models/stock_picking_note.py (hunk 3 above)
Comment (santostelmo) [reply stock_picking_partner_note/models/stock_picking_note.py]: Thank you it's also an option indeed

**Comment by santostelmo** [comment]
> Code review.
> 
> @santostelmo could you also add some tests ?

Unit test added

**Comment by cyrilmanuel** [comment]
@rousseldenis Little ping, is it good for you ?

**Review by cyrilmanuel — APPROVED** [review]
LGTM

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**